hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/expression/json/JsonQueryIntegrationTest.java

```diff
@@ -0,0 +1,182 @@
+        assertEquals(new HazelcastJsonValue("[]"),
+                querySingleValue("SELECT JSON_QUERY(this, '$' EMPTY ARRAY ON EMPTY) AS c1 FROM test WHERE __key = 1"));
+        assertEquals(new HazelcastJsonValue("{}"),
+                querySingleValue("SELECT JSON_QUERY(this, '$' EMPTY OBJECT ON EMPTY) AS c1 FROM test WHERE __key = 1"));
+        assertThatThrownBy(() -> query("SELECT JSON_QUERY(this, '$' ERROR ON EMPTY) AS c1 FROM test WHERE __key = 1"))
+                .isInstanceOf(HazelcastSqlException.class)
+                .hasMessageContaining("Empty JSON object");
+
+        assertNull(querySingleValue("SELECT JSON_QUERY(this, '$' NULL ON ERROR) AS c1 FROM test WHERE __key = 2"));
+        assertEquals(new HazelcastJsonValue("[]"),
+                querySingleValue("SELECT JSON_QUERY(this, '$' EMPTY ARRAY ON ERROR) AS c1 FROM test WHERE __key = 2"));
+        assertEquals(new HazelcastJsonValue("{}"),
+                querySingleValue("SELECT JSON_QUERY(this, '$' EMPTY OBJECT ON ERROR) AS c1 FROM test WHERE __key = 2"));
+        assertThatThrownBy(() -> query("SELECT JSON_QUERY(this, '$' ERROR ON ERROR) AS c1 FROM test WHERE __key = 2"))
+                .isInstanceOf(HazelcastSqlException.class)
+                .hasMessageContaining("JSON_QUERY failed");
+    }
+
+    @Test
+    public void when_defaultWrapperBehaviorIsSpecified_queryWorks() {
+        initComplexObject();
+        execute("CREATE MAPPING test TYPE IMap OPTIONS ('keyFormat'='bigint', 'valueFormat'='json')");
+        assertEquals("[1,\"2\",3,{\"t\":1}]",
+                querySingleValue("SELECT JSON_QUERY(this, '$[0]') FROM test").toString());
+        assertEquals("{\"t\":1}",
+                querySingleValue("SELECT JSON_QUERY(this, '$[1]') FROM test").toString());
+        assertThatThrownBy(() -> querySingleValue("SELECT JSON_QUERY(this, '$[2]' ERROR ON ERROR) FROM test"))
+                .isInstanceOf(HazelcastSqlException.class)
+                .hasMessageContaining("JSON_QUERY result is not an array or object");
+    }
+
+    @Test
+    public void when_noArrayWrapperSpecified_queryWorks() {
+        initComplexObject();
+        execute("CREATE MAPPING test TYPE IMap OPTIONS ('keyFormat'='bigint', 'valueFormat'='json')");
+
+        assertEquals("[1,\"2\",3,{\"t\":1}]",
+                querySingleValue("SELECT JSON_QUERY(this, '$[0]' WITHOUT ARRAY WRAPPER) FROM test")
+                        .toString());
+        assertEquals("{\"t\":1}",
+                querySingleValue("SELECT JSON_QUERY(this, '$[1]' WITHOUT ARRAY WRAPPER) FROM test")
+                        .toString());
+        assertThatThrownBy(() -> querySingleValue("SELECT JSON_QUERY(this, '$[2]' WITHOUT ARRAY WRAPPER ERROR ON ERROR) "
+                        + "FROM test"))
+                .isInstanceOf(HazelcastSqlException.class)
+                .hasMessageContaining("JSON_QUERY result is not an array or object");
```

Comment:
This test case added here fails:
```java
        assertThatThrownBy(() -> querySingleValue("SELECT JSON_QUERY(this, '$[*]' WITHOUT ARRAY WRAPPER ERROR ON ERROR) "
                        + "FROM test"))
                .isInstanceOf(HazelcastSqlException.class)
                .hasMessageContaining("result cannot be returned without array wrapper");
```
Reasoning: the result of the above json path is not an array, but rather a sequence of elements which need to be wrapped. Oracle also throws an error in this case.